### ISSUE TITLE
DEFMFUN creates $NAME and $NAME-IMPL, contradicting documentation

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1907]
Imported from SourceForge on 2024-07-05 10:28:02
Created by **[mkoeppe](https://sourceforge.net/u/mkoeppe/)** on 2020-07-20 04:05:40
Original: https://sourceforge.net/p/maxima/bugs/3643

---

The documentation of DEFMFUN claims the following:
;; If the function name NAME starts with #\$ we check the number of
;; arguments.  In this case, two functions are created: NAME and
;; NAME-IMPL (without the leading $). 

However, actually the leading $ is not removed. This causes 
$APROPOS to leak these details to the user. Observed in Maxima 5.44.0

See https://trac.sagemath.org/ticket/30063#comment:17

~~~
$ ./sage -maxima
;;; Loading #P"/Users/mkoeppe/s/sage/sage-rebasing/local/lib/ecl/sb-bsd-sockets.fas"
;;; Loading #P"/Users/mkoeppe/s/sage/sage-rebasing/local/lib/ecl/sockets.fas"
Maxima 5.44.0 http://maxima.sourceforge.net
using Lisp ECL 20.4.24
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) apropos("gcd");
(%o1) [ef_gcd, ef_gcd-impl, ef_gcdex, ef_gcdex-impl, ezgcd, ezgcd-impl, gcd, 
            gcd-impl, gcdex, gf_gcd, gf_gcd-impl, gf_gcdex, gf_gcdex-impl, gcd]
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:28:04
Created by **[rtoy](https://sourceforge.net/u/rtoy/)** on 2020-09-16 23:25:17
Original: https://sourceforge.net/p/maxima/bugs/3643/#e9ed

---

This change looks fine to me.  I don't see any reason why the impl needs a leading $.  It's an internal implementation detail that the user doesn't need to see.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:28:07
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-09-20 00:13:11
Original: https://sourceforge.net/p/maxima/bugs/3643/#c210

---

- **labels**:  --> lisp
- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:28:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-09-20 00:13:11
Original: https://sourceforge.net/p/maxima/bugs/3643/#5cce

---

Fixed by commit 54a22fb. Closing this ticket, thank you for the bug report, I appreciate it.
